### PR TITLE
Fix default trace id

### PR DESCRIPTION
**Pull request by vincent0426**
# Fix external workflow trace ID defaulting to zeros

## Summary
Fixes the issue where external workflow trace IDs were defaulting to "00000000-0000-0000-0000-000000000000" instead of proper UUIDs.

**Root Cause**: The `execution_context()` function was using `int(prev_context.trace_id)` to determine when to preserve vs. generate trace IDs, but this logic didn't properly handle the relationship between nested contexts and fresh contexts.

**Solution**: Updated the trace ID logic to:
1. Use `parent_context` presence as the primary indicator for nested contexts
2. Preserve existing trace_id when in nested contexts (and trace_id is not zero)
3. Allow explicit trace_id when starting fresh contexts
4. Generate new UUID as fallback

## Review & Testing Checklist for Human
- [ ] **Test external workflow execution end-to-end** - Verify that external workflows now generate proper UUID trace IDs instead of zeros
- [ ] **Verify nested context behavior** - Test that trace IDs are preserved correctly when contexts are nested (e.g., subworkflows, node executions)
- [ ] **Check conditional logic edge cases** - Review the 3-condition logic in `execution_context()` for potential edge cases or unintended behavior
- [ ] **Run the originally failing tests** - Confirm that the 4 tests mentioned in CI (test_context_trace_and_parent, inline_prompt_node tests, etc.) now pass consistently

### Notes
- The fix addresses both the symptom (zero trace IDs) and ensures proper trace ID propagation in nested contexts
- Only minimal test changes were made - main validation comes from existing test suite passing
- CI tests are now passing, but end-to-end external workflow testing is recommended

---
**Link to Devin run**: https://app.devin.ai/sessions/0d8fa169a5774d218abd6599a4fcfd39  
**Requested by**: @vincent0426 (vincent@vellum.ai)

### CODE REVIEW

File: src/vellum/workflows/context.py [hunk 1]
```diff
@@ -60,7 +60,14 @@ def execution_context(
...
     prev_context = get_execution_context()
-    set_trace_id = prev_context.trace_id if int(prev_context.trace_id) else trace_id or uuid4()
+    if prev_context.parent_context is not None and prev_context.trace_id != UUID(
+        "00000000-0000-0000-0000-000000000000"
+    ):
+        set_trace_id = prev_context.trace_id
+    elif trace_id is not None:
+        set_trace_id = trace_id
+    else:
+        set_trace_id = uuid4()
```
Comment (dvargasfuertes) [inline src/vellum/workflows/context.py]: [blocking] I would solve this at the `WorkflowContext` level so that it's specific to when we add an `ExternalParentContext`

**Review by marlenc3 — APPROVED** [review]

**Review by dvargas92495 — APPROVED** [review]
